src/Api/ApiPlayground.tsx

```diff
@@ -0,0 +1,152 @@
+import clsx from "clsx";
+import { ReactNode, useState } from "react";
+import {
+  getMethodBgColor,
+  getMethodTextColor,
+  getMethodBorderColor,
+  getMethodBgHoverColor,
+} from "../utils/apiPlaygroundColors";
+import { ApiInput } from "./ApiInput";
+import { ApiInputValue, ParamGroup, RequestMethods } from "./types";
+
+export function ApiPlayground({
+  method,
+  paramGroups,
+  paramValues,
+  isSendingRequest,
+  onChangeParam,
+  onSendRequest,
+  header,
+  response,
+}: {
+  /** Request method. */
+  method: RequestMethods;
+
+  /** Array of param groups to show as tabs for input. */
+  paramGroups: ParamGroup[];
+
+  /** Values to show in the ApiInputs. Key is the param group name. */
+  paramValues: Record<string, Record<string, any>>;
+
+  /** Whether you are currently sending a request.
+   *  The Send Request button is disabled and the response is hidden while this is true. */
+  isSendingRequest: boolean;
+
+  /** Callback when the user changes a parameter's value. */
+  onChangeParam: (
+    paramGroupName: string,
+    parentInputs: string[],
+    paramName: string,
+    paramValue: ApiInputValue
+  ) => void;
+
+  /** Callback when the user clicks the Send Request button. */
+  onSendRequest: () => void;
+
+  /** Header to show above parameter inputs. */
+  header?: ReactNode;
+
+  /** Response to show underneath the playground.
+   *  This component does not automatically syntax highlight code. */
+  response?: ReactNode;
+}) {
+  // const onChangeParam = (
+  //   paramGroup: string,
+  //   param: string,
+  //   value: ApiInputValue,
+  //   path: string[]
+  // ) => {
+  //   const newParamGroup = {
+  //     ...inputData[paramGroup],
+  //     ...set(inputData[paramGroup], [...path, param], value),
+  //   };
+  //   setInputData({ ...inputData, [paramGroup]: newParamGroup });
+  // };
```

Comment:
What is this?

Reply:
I'm working on using these today! I'm not publishing to npm yet so I'm going to merge for now.